File: 58. Length of Last Word/last.kt
class Solution{

fun lengthOfLastWord(s: String): Int{

var i=s.length-1
var length=0

 // skip trailing spaces
        while (i >= 0 && s[i] == ' ') {
            i--
        }

        // count last word
        while (i >= 0 && s[i] != ' ') {
            length++
            i--
        }

        return length
}
}

fun main(){

val s=readln().toString()
val obj=Solution()
println(obj.lengthOfLastWord(s))
}

File: 58. Length of Last Word/last.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution{
public:

int lengthOfLastWord(string s){

    int cnt=0;
    int i=s.size()-1;

    while(i>=0 && s[i]==' '){
    i--;
    }

    while(i>=0 && s[i]!=' '){
    cnt++;
    i--;
    }

    return cnt;



}
};

int main(){

string s;
cin>>s;

Solution obj;
cout<<obj.lengthOfLastWord(s);
}
